server/src/charts/charts.service.ts

```diff
@@ -0,0 +1,100 @@
+import { forwardRef, Inject, Injectable } from '@nestjs/common'
+import { InjectRepository } from '@nestjs/typeorm'
+import { User } from 'src/auth/user.types'
+import { QueriesService } from 'src/queries/queries.service'
+import { Chart } from './chart.entity'
+import { ChartRepository } from './chart.repository'
+import { CreateChartDto, FilterChartDto, UpdateChartDto } from './charts.dto'
+
+@Injectable()
+export class ChartsService {
+  constructor(
+    @InjectRepository(ChartRepository)
+    private chartsRepository: ChartRepository,
+    @Inject(forwardRef(() => QueriesService))
+    private queriesService: QueriesService,
+  ) {}
+
+  /**
+   * Returns all the charts
+   *
+   * @param user
+   * @param query
+   */
+  getAllCharts(user: User, filterChartDto: FilterChartDto): Promise<Chart[]> {
+    const { query } = filterChartDto
```

Comment:
We can merge these two if statements as 

```typescript
const where = { uid: user.uid, query: query ? { id: query } : undefined}
return this.chartsRepository.find({ order: { createdAt: 'ASC' }, where })
```

Reply:
sending undefined to query in find method is not working